thriftbp/server.go

```diff
@@ -190,3 +195,19 @@ var (
 	_ baseplate.Server = impl{}
 	_ baseplate.Server = (*impl)(nil)
 )
+
+func GetThriftServiceName(processor thrift.TProcessor) string {
+	t := reflect.TypeOf(processor)
+	if t == nil {
+		return "unknown"
+	}
+	value := t.Elem()
+	if value == nil {
+		return "unknown"
+	}
+
+	return fmt.Sprintf("%s.%s",
+		filepath.Base(value.PkgPath()),
```

Comment:
why do we use `filepath` here? `filepath` is the one that uses `/` for *nix and `\` for windows, I don't think that's how go package path works. it should always be `/` (as go package path is kind of url based) so we should use `path` instead.

also do you have an example/test what this is supposed to get? I'm not sure why we infer this from go package path instead of set explicitly.

Reply:
Good call on filepath. Ill push that fix. I added a test to show what the expected output is. Im pulling the last part of the go path because that is usually defined by either go package set in the thrift file or the name of the thrift file itself. I then append the name of the processor with the processor suffix stripped since the thrift compiler matches that to the service schema name. Its not perfect. But I think its going to give us more consistent results. Especially since we can do something similar in the thrift client. This option should get us the closest to proper package and service names like grpc. But Im open to alternatives.